packages/vscode/src/plugins/nextjs/addJSDoc.ts

```diff
@@ -0,0 +1,88 @@
+import {
+  ArrowFunction,
+  FunctionDeclaration,
+  SourceFile,
+  TypeGuards,
+} from 'ts-morph'
+import { NextFunctionName, NextFunctionType } from './constants'
+
+function buildJSDocType(base: string, paramName: string) {
+  return `{import('next').${base} } ${paramName}`
+}
+function addType(
+  node: ArrowFunction | FunctionDeclaration,
+  foundNextFunctions: NextFunctionType,
+) {
+  const functionNames = Object.keys(foundNextFunctions) as NextFunctionName[]
+  if (functionNames && functionNames.length !== 0) {
+    const params = node.getParameters()
+    const param = params[0]
+    const typeString = foundNextFunctions[functionNames[0]].type
+    const jsDocType = buildJSDocType(typeString, 'props')
+    if (
+      TypeGuards.isFunctionDeclaration(node) &&
+      node.getJsDocs().length === 0
+    ) {
+      if (!param) {
+        node.addParameter({
+          name: 'props',
+        })
+      }
+      node.addJsDoc({
+        tags: [
+          {
+            tagName: 'param',
+            text: jsDocType,
+          },
+        ],
+      })
+    }
+  }
+}
+
+export function addJSDoc(
+  sourceFile: SourceFile,
+  foundNextFunctions: NextFunctionType,
+) {
+  const defaultExportSymbol = sourceFile.getDefaultExportSymbol()
+
+  if (defaultExportSymbol) {
```

Comment:
What does that mean? Is it supposed to never be undefined? If not, there should be some log here.

Reply:
There might not be a default export in the file. If there isn't then just skip it